src/main/java/com/mitrais/chipper/temankondangan/backendapps/service/impl/ProfileServiceImpl.java

```diff
@@ -12,186 +29,158 @@
-        this.userRepository = userRepository;
-        this.profileRepository = profileRepository;
-        this.passwordEncoder = passwordEncoder;
-    }
-
-    private static final String DEFAULT_IMAGE = "image/defaultprofile.jpg";
-
-    @Override
-    public Profile create(CreateProfileWrapper wrapper) {
-        User user = userRepository.findByEmail(wrapper.getEmail())
-                .orElseThrow(() -> new ResourceNotFoundException("User", "email", wrapper.getEmail()));
-
-        // check dob valid
-        LocalDate dob;
-        DateTimeFormatter df = DateTimeFormatter.ofPattern("dd-MM-uuuu").withResolverStyle(ResolverStyle.STRICT);
-        try {
-            dob = LocalDate.parse(wrapper.getDob(), df);
-        } catch (Exception e) {
-            throw new BadRequestException("Error: Date not valid!");
-        }
-
-        // check age over 18
-        if (Period.between(dob, LocalDate.now()).getYears() < 18) {
-            throw new BadRequestException("Error: Age should not under 18!");
-        }
-
-        Profile profile = profileRepository.findByUserId(user.getUserId())
-                .orElse(new Profile());
-        profile.setUser(user);
-        profile.setFullName(wrapper.getFullname());
-        profile.setDob(dob);
-        profile.setGender(wrapper.getGender());
-        profile.setDataState(DataState.ACTIVE);
-        return profileRepository.save(profile);
-    }
-
-    @Override
-    public Profile update(Long userId, ProfileUpdateWrapper wrapper) {
-        Profile profile = profileRepository.findByUserId(userId)
-                .orElseThrow(() -> new BadRequestException("Error: User not found!"));
-
-        byte[] image = null;
-        String fileName = "";
-
-        String[] allowedFormatImage = {"jpeg", "png", "jpg"};
-        List<String> allowedFormatImageList = Arrays.asList(allowedFormatImage);
-
-        // if null or if not select anything
-        if (wrapper.getImage() == null && profile.getPhotoProfile() == null) {
-            image = readBytesFromFile(DEFAULT_IMAGE);
-            fileName = DEFAULT_IMAGE.split("/")[1];
-        } else if (wrapper.getImage() != null && !StringUtils.isEmpty(wrapper.getImage().getOriginalFilename())) {
-            // throw error if image format is not allowed
-            String[] imageFormat = wrapper.getImage().getOriginalFilename().split("\\.");
-            if (!allowedFormatImageList.contains(imageFormat[imageFormat.length - 1])) {
-                throw new BadRequestException("Error: Image format not allowed!");
-            }
-            try {
-                image = wrapper.getImage().getBytes();
-                fileName = wrapper.getImage().getOriginalFilename().replaceAll("\\s+", "-");
-            } catch (IOException e) {
-                throw new BadRequestException(e.getMessage());
-            }
-        }
-
-        profile.setPhotoProfile(image);
-        profile.setPhotoProfileFilename(fileName);
```

Comment:
I think you forget to set the filename of photo profile.

Reply:
oh that's right!